internal/daemonset/daemonset.go

```diff
@@ -0,0 +1,199 @@
+package daemonset
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"github.com/test-network-function/cnf-certification-test/internal/clientsholder"
+	"github.com/test-network-function/cnf-certification-test/pkg/provider"
+	v1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	imageWithVersion                   = "quay.io/testnetworkfunction/debug-partner:latest"
+	timeout                            = 5 * time.Minute
+	daemonsetDeletionCheckRetryInteval = 5 * time.Second
+	nodeExporter                       = "node-exporter"
+	containerName                      = "container-00"
+	debug                              = "debug"
+)
+
+// Delete daemon set
+func DeleteDaemonSet(daemonSetName, namespace string) error {
+	logrus.Infof("Deleting daemon set %s", daemonSetName)
+	deletePolicy := metav1.DeletePropagationForeground
+	client := clientsholder.GetClientsHolder().K8sClient
+	err := client.AppsV1().DaemonSets(namespace).Delete(context.TODO(), daemonSetName, metav1.DeleteOptions{
+		PropagationPolicy: &deletePolicy})
+	if err != nil {
+		return fmt.Errorf("daemonset %s deletion failed: %w", daemonSetName, err)
+	}
+	allPodsRemoved := false
+	for start := time.Now(); time.Since(start) < timeout; {
+		pods, err := client.CoreV1().Pods(namespace).List(context.TODO(), metav1.ListOptions{LabelSelector: "test-network-function.com/app=debug"})
+		if err != nil {
+			return fmt.Errorf("failed to get pods, err: %s", err)
+		}
+
+		if len(pods.Items) == 0 {
+			allPodsRemoved = true
+			break
+		}
+		allPodsRemoved = false
```

Comment:
You don't need this assignment to false on each iteration, as it's already initialized in line 35 and the only place where it could change is in line 43, right before leaving the loop.